### PR TITLE
Cleanup orphaned managed git repos in deploy tests

### PR DESCRIPTION
**Pull request by k-anshul**
The deployments tests do not delete managed repos. The repos are currently deleted via a background worker only after  an inactive period of 7 days.

**Checklist:**
- [x] Covered by tests
- [x] Ran it and it works as intended
- [x] Reviewed the diff before requesting a review
- [x] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [ ] I'm proud of this work!

### CODE REVIEW

File: .claude/CLAUDE.md [hunk 1]
```diff
@@ -69,6 +69,7 @@ General rules for writing Go code:
 - Avoid short utility functions that are only used once; it is usually more readable to inline these in their parent function.
 - Avoid variables that only serve as aliases unless there's a large readability improvement (i.e. the expression is very long and used several times); for example, instead of `db := table.Database`, just reference `table.Database` directly.
 - Use the standard library `errors` (not `github.com/pkg/errors`).
+- Prefer `require.NoError(...)` instead of `panic` in tests.
```
Comment (k-anshul) [inline .claude/CLAUDE.md:72]: I often see it using `panic` in utility functions.

File: cli/cmd/project/deployment/deployments_test.go [hunk 2]
```diff
@@ -140,6 +144,10 @@ olap_connector: duckdb`,
 	result = u1.Run(t, "project", "deploy", "--interactive=false", "--org=branch-change-test", "--project=branch-test", "--path="+tempDir)
 	require.Equal(t, 0, result.ExitCode, result.Output)
 
+	t.Cleanup(func() {
+		adm.DeleteManagedRepo(t, tempDir)
+	})
```
Comment (begelundmuller) [inline cli/cmd/project/deployment/deployments_test.go]: Not sure, but did you consider adding a `t.Cleanup` inside of `testadmin.NewWithOptionalRuntime` that uses `adm.DB` to find every managed repo and delete it? Just so the individual test cases don't need to specify this.
